m2cgen/assemblers/fallback_expressions.py

```diff
@@ -0,0 +1,43 @@
+from m2cgen import ast
+from m2cgen.assemblers import utils
+
+
+def tanh(expr):
+    expr.to_reuse = True
```

Comment:
I'm a little bit concerned by the fact that here we implicitly alter the state of the expression passed as an input argument. I'm worrying that if we adopt this kind of practice going forward this may lead to some unexpected behaviors in future. What do you say if instead we introduce some kind of identity expression, eg. `IdExpr`. It may transparently wrap the original expression but have a reuse flag set true. What do you think of this concern in general?

Reply:
> What do you say if instead we introduce some kind of identity expression, eg. `IdExpr`.

Are reading my thoughts?! 😄 
The first thing that came to my mind was to use some dummy wrapper for the passed expression. And I searched `ast.py` for any but unfortunately didn't succeed in it.

I think it's better to add `IdExpr` in a separate PR to not overcomplicate this one. I'll prepare it.